packages/server/src/api/controllers/record.js

```diff
@@ -231,3 +235,55 @@ async function validate({ instanceId, modelId, record, model }) {
   }
   return { valid: Object.keys(errors).length === 0, errors }
 }
+
+function coerceRecordValues(record, model) {
+  for (let [key, value] of Object.entries(record)) {
+    const field = model.schema[key]
+    if (!field) continue
+    const mapping = Object.prototype.hasOwnProperty.call(
+      TYPE_TRANSFORM_MAP[field.type],
+      value
+    )
+      ? TYPE_TRANSFORM_MAP[field.type][value]
+      : TYPE_TRANSFORM_MAP[field.type].parse
+
+    record[key] = typeof mapping === "function" ? mapping(value) : mapping
+  }
+}
+
+const TYPE_TRANSFORM_MAP = {
+  string: {
+    "": "",
+    [null]: "",
+    [undefined]: undefined,
+    parse: s => s,
+  },
+  number: {
+    "": null,
+    [null]: null,
+    [undefined]: undefined,
+    parse: n => parseFloat(n),
+  },
+  datetime: {
+    "": null,
+    [undefined]: undefined,
+    [null]: null,
+    parse: d => d,
+  },
+  attachment: {
+    "": [],
+    [null]: [],
+    [undefined]: undefined,
+    parse: a => a,
+  },
+  boolean: {
+    "": null,
+    [null]: null,
+    [undefined]: undefined,
+    parse: b => {
```

Comment:
is the `parse` function necessary in any of these? can we not just do:

```
boolean: {
     "": null,
     [null]: null,
     [undefined]: undefined,
     "true": true,
     "false": false
}
```

Reply:
i removed it for most. The reason it is there is because of parsing numbers from strings... "123" > 123 . Looked into using ValidateJS to do this, like datetime, but couldn't see a way of doing so.